travel/src/main/java/com/tx/travel/service/DailyAllowanceService.java

```diff
@@ -0,0 +1,88 @@
+package com.tx.travel.service;
+
+import com.tx.travel.mapper.DailyAllowanceMapper;
+import com.tx.travel.model.DailyAllowance;
+import com.tx.travel.payload.request.DailyAllowanceRequest;
+import com.tx.travel.payload.response.DailyAllowanceResponse;
+import com.tx.travel.repository.DailyAllowanceRepository;
+import com.tx.travel.service.exception.DailyAllowanceNotFoundException;
+import com.tx.travel.service.exception.RegionAlreadyExistsException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+@Service
+public class DailyAllowanceService {
+
+    private final DailyAllowanceMapper dailyAllowanceMapper;
+    private final DailyAllowanceRepository dailyAllowanceRepository;
+
+    public DailyAllowanceService(final DailyAllowanceRepository dailyAllowanceRepository, final DailyAllowanceMapper dailyAllowanceMapper) {
+        this.dailyAllowanceMapper = dailyAllowanceMapper;
+        this.dailyAllowanceRepository = dailyAllowanceRepository;
+    }
+
+    public List<DailyAllowanceResponse> getAllDailyAllowances() {
+        List<DailyAllowanceResponse> dailyAllowanceResponses = new CopyOnWriteArrayList<>();
```

Comment:
here there should be another layer of mapping, so controller uses DailyAllowanceResponse, service uses domain object DailyAllowance and repository uses DailyAllowanceDao, consult with @DusanBrankovic  about this

Reply:
☝️